Feature/filter samples by value

zsloan opened the pull request:
#### Description
This PR adds the option to filter sample values, SEs, and case attributes (the extra cofactor columns) using >, <, ≥, and ≤.

It seems to work correctly, but I want to somehow simplify the interface because the "Transform and Filter" tab has too many options clustered in it now.

#### How should this be tested?
Load any trait page (like the one below), go to the "Transform and Filter Data" tab, and try the "Filter samples by" option.

https://genenetwork.org/show_trait?trait_id=10011&dataset=BXD-LongevityPublish

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)

#### Questions

zsloan committed:
Added html/css for new "Filter Samples by Value" feature and changed a couple class names to make more sense for being shared between both this feature and "Block Samples by Index"

zsloan committed:
Added JS for Filter Samples by Value feature and removed a few unused lines

zsloan committed:
- Fixed the "Block by Attribute Value" feature
- Changed the "Block by Attribute Value" feature and the "Filter by Value" feature to only work on columns that are numeric or have a small number of distinct values (respectively)

robwwilliams commented:
@Zach Farrow <zach.farrow@novogene.com>
I went to this URL after closing and then opening Firefox

http://gn2-zach.genenetwork.org/show_trait?trait_id=10002&dataset=BXD-LongevityPublish


Is this the interface I should see?
No obvious place to enter text except the Block Index field where  the
index range values are entered.

I did clear cache. Hmm.

[image: Screen Shot 2020-11-25 at 1.12.34 PM.png]

[image: Screen Shot 2020-11-25 at 1.09.53 PM.png]

Send an image of what I should see.


On Wed, Nov 25, 2020 at 12:49 PM zsloan <notifications@github.com> wrote:

> Description
>
> This PR adds the option to filter sample values, SEs, and case attributes
> (the extra cofactor columns) using >, <, ≥, and ≤.
>
> It seems to work correctly, but I want to somehow simplify the interface
> because the "Transform and Filter" tab has too many options clustered in it
> now.
> How should this be tested?
>
> Load any trait page (like the one below), go to the "Transform and Filter
> Data" tab, and try the "Filter samples by" option.
>
>
> https://genenetwork.org/show_trait?trait_id=10011&dataset=BXD-LongevityPublish
> Any background context you want to provide?
>
> N/A
> What are the relevant pivotal tracker stories?
>
> N/A
> Screenshots (if appropriate) Questions
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/genenetwork/genenetwork2/pull/490
> Commit Summary
>
>    - Added html/css for new "Filter Samples by Value" feature and changed
>    a couple class names to make more sense for being shared between both this
>    feature and "Block Samples by Index"
>    - Added JS for Filter Samples by Value feature and removed a few
>    unused lines
>    - - Fixed the "Block by Attribute Value" feature
>
> File Changes
>
>    - *M* wqflask/wqflask/static/new/css/show_trait.css
>    <https://github.com/genenetwork/genenetwork2/pull/490/files#diff-0b48c6cbdf6435d619eeedf5ccf974b1e5a6cde84338005b164afb27d573c246>
>    (4)
>    - *M* wqflask/wqflask/static/new/javascript/show_trait.js
>    <https://github.com/genenetwork/genenetwork2/pull/490/files#diff-3edfd5623c8d344941dcfe7356a4d540334f4c6c67ff97a47a82e6c1f8abd9f9>
>    (138)
>    - *M* wqflask/wqflask/templates/show_trait_transform_and_filter.html
>    <https://github.com/genenetwork/genenetwork2/pull/490/files#diff-f47c2ec793606e1eb79dfb5ce3d0295e70db9e7e55f751ffa3292c6cdce2c71c>
>    (59)
>
> Patch Links:
>
>    - https://github.com/genenetwork/genenetwork2/pull/490.patch
>    - https://github.com/genenetwork/genenetwork2/pull/490.diff
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/490>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7CJLNHDDX4QKX5MRXLSRVGSVANCNFSM4UCYV53Q>
> .
>


-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

robwwilliams commented:
Trying. Do not see "Filter Samples" by attribute function yet.

On Wed, Nov 25, 2020 at 12:49 PM zsloan <notifications@github.com> wrote:

> @zsloan <https://github.com/zsloan> requested your review on: #490
> <https://github.com/genenetwork/genenetwork2/pull/490> Feature/filter
> samples by value.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/490#event-4039449788>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7GY36TXV5GRTGH4CQLSRVGSXANCNFSM4UCYV53Q>
> .
>


-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

zsloan commented:
No, my branch isn't using the branch with this change in it right now. Not
sure how to make it available while working on something else.

I can temporarily change to that branch, though. I'll change it for now and
it'll be up for the next hour.

(By the way, you apparently used the wrong Zach's e-mail in that @)

On Wed, Nov 25, 2020 at 1:20 PM robwwilliams <notifications@github.com>
wrote:

> @Zach Farrow <zach.farrow@novogene.com>
> I went to this URL after closing and then opening Firefox
>
>
> http://gn2-zach.genenetwork.org/show_trait?trait_id=10002&dataset=BXD-LongevityPublish
>
>
> Is this the interface I should see?
> No obvious place to enter text except the Block Index field where the
> index range values are entered.
>
> I did clear cache. Hmm.
>
> [image: Screen Shot 2020-11-25 at 1.12.34 PM.png]
>
> [image: Screen Shot 2020-11-25 at 1.09.53 PM.png]
>
> Send an image of what I should see.
>
>
> On Wed, Nov 25, 2020 at 12:49 PM zsloan <notifications@github.com> wrote:
>
> > Description
> >
> > This PR adds the option to filter sample values, SEs, and case attributes
> > (the extra cofactor columns) using >, <, ≥, and ≤.
> >
> > It seems to work correctly, but I want to somehow simplify the interface
> > because the "Transform and Filter" tab has too many options clustered in
> it
> > now.
> > How should this be tested?
> >
> > Load any trait page (like the one below), go to the "Transform and Filter
> > Data" tab, and try the "Filter samples by" option.
> >
> >
> >
> https://genenetwork.org/show_trait?trait_id=10011&dataset=BXD-LongevityPublish
> > Any background context you want to provide?
> >
> > N/A
> > What are the relevant pivotal tracker stories?
> >
> > N/A
> > Screenshots (if appropriate) Questions
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/genenetwork/genenetwork2/pull/490
> > Commit Summary
> >
> > - Added html/css for new "Filter Samples by Value" feature and changed
> > a couple class names to make more sense for being shared between both
> this
> > feature and "Block Samples by Index"
> > - Added JS for Filter Samples by Value feature and removed a few
> > unused lines
> > - - Fixed the "Block by Attribute Value" feature
> >
> > File Changes
> >
> > - *M* wqflask/wqflask/static/new/css/show_trait.css
> > <
> https://github.com/genenetwork/genenetwork2/pull/490/files#diff-0b48c6cbdf6435d619eeedf5ccf974b1e5a6cde84338005b164afb27d573c246
> >
> > (4)
> > - *M* wqflask/wqflask/static/new/javascript/show_trait.js
> > <
> https://github.com/genenetwork/genenetwork2/pull/490/files#diff-3edfd5623c8d344941dcfe7356a4d540334f4c6c67ff97a47a82e6c1f8abd9f9
> >
> > (138)
> > - *M* wqflask/wqflask/templates/show_trait_transform_and_filter.html
> > <
> https://github.com/genenetwork/genenetwork2/pull/490/files#diff-f47c2ec793606e1eb79dfb5ce3d0295e70db9e7e55f751ffa3292c6cdce2c71c
> >
> > (59)
> >
> > Patch Links:
> >
> > - https://github.com/genenetwork/genenetwork2/pull/490.patch
> > - https://github.com/genenetwork/genenetwork2/pull/490.diff
> >
> > —
> > You are receiving this because your review was requested.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/genenetwork/genenetwork2/pull/490>, or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AC55V7CJLNHDDX4QKX5MRXLSRVGSVANCNFSM4UCYV53Q
> >
> > .
> >
>
>
> --
> Rob
>
> Robert W. Williams, Ph.D.
> Chair: Department of Genetics, Genomics and Informatics
> University of Tennessee Health Science Center
> 71 S Manassas St, Memphis TN 38103-3310, TSRB Room 405
> Tele/Cell: 901 604 4752
> EMAIL: rwilliams@uthsc.edu or labwilliams@gmail.com
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/490#issuecomment-733903592>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGDILPUBMSRVE324STLSRVKHVANCNFSM4UCYV53Q>
> .
>

zsloan commented:
I kind of wonder if it might make sense to have a "Rob testing site" or
something where changes up for "Rob review" can be used.

In the past this was less of an issue because I wasn't really using git
correctly and was making all changes on the same local branch, but now that
I'm actually separating them into separate branches it's difficult for Rob
to test them if I'm currently using a different branch than the one with
the changes.

On Wed, Nov 25, 2020 at 1:33 PM Zachary Sloan <zachary.a.sloan@gmail.com>
wrote:

> No, my branch isn't using the branch with this change in it right now. Not
> sure how to make it available while working on something else.
>
> I can temporarily change to that branch, though. I'll change it for now
> and it'll be up for the next hour.
>
> (By the way, you apparently used the wrong Zach's e-mail in that @)
>
> On Wed, Nov 25, 2020 at 1:20 PM robwwilliams <notifications@github.com>
> wrote:
>
>> @Zach Farrow <zach.farrow@novogene.com>
>> I went to this URL after closing and then opening Firefox
>>
>>
>> http://gn2-zach.genenetwork.org/show_trait?trait_id=10002&dataset=BXD-LongevityPublish
>>
>>
>> Is this the interface I should see?
>> No obvious place to enter text except the Block Index field where the
>> index range values are entered.
>>
>> I did clear cache. Hmm.
>>
>> [image: Screen Shot 2020-11-25 at 1.12.34 PM.png]
>>
>> [image: Screen Shot 2020-11-25 at 1.09.53 PM.png]
>>
>> Send an image of what I should see.
>>
>>
>> On Wed, Nov 25, 2020 at 12:49 PM zsloan <notifications@github.com> wrote:
>>
>> > Description
>> >
>> > This PR adds the option to filter sample values, SEs, and case
>> attributes
>> > (the extra cofactor columns) using >, <, ≥, and ≤.
>> >
>> > It seems to work correctly, but I want to somehow simplify the interface
>> > because the "Transform and Filter" tab has too many options clustered
>> in it
>> > now.
>> > How should this be tested?
>> >
>> > Load any trait page (like the one below), go to the "Transform and
>> Filter
>> > Data" tab, and try the "Filter samples by" option.
>> >
>> >
>> >
>> https://genenetwork.org/show_trait?trait_id=10011&dataset=BXD-LongevityPublish
>> > Any background context you want to provide?
>> >
>> > N/A
>> > What are the relevant pivotal tracker stories?
>> >
>> > N/A
>> > Screenshots (if appropriate) Questions
>> > ------------------------------
>> > You can view, comment on, or merge this pull request online at:
>> >
>> > https://github.com/genenetwork/genenetwork2/pull/490
>> > Commit Summary
>> >
>> > - Added html/css for new "Filter Samples by Value" feature and changed
>> > a couple class names to make more sense for being shared between both
>> this
>> > feature and "Block Samples by Index"
>> > - Added JS for Filter Samples by Value feature and removed a few
>> > unused lines
>> > - - Fixed the "Block by Attribute Value" feature
>> >
>> > File Changes
>> >
>> > - *M* wqflask/wqflask/static/new/css/show_trait.css
>> > <
>> https://github.com/genenetwork/genenetwork2/pull/490/files#diff-0b48c6cbdf6435d619eeedf5ccf974b1e5a6cde84338005b164afb27d573c246
>> >
>> > (4)
>> > - *M* wqflask/wqflask/static/new/javascript/show_trait.js
>> > <
>> https://github.com/genenetwork/genenetwork2/pull/490/files#diff-3edfd5623c8d344941dcfe7356a4d540334f4c6c67ff97a47a82e6c1f8abd9f9
>> >
>> > (138)
>> > - *M* wqflask/wqflask/templates/show_trait_transform_and_filter.html
>> > <
>> https://github.com/genenetwork/genenetwork2/pull/490/files#diff-f47c2ec793606e1eb79dfb5ce3d0295e70db9e7e55f751ffa3292c6cdce2c71c
>> >
>> > (59)
>> >
>> > Patch Links:
>> >
>> > - https://github.com/genenetwork/genenetwork2/pull/490.patch
>> > - https://github.com/genenetwork/genenetwork2/pull/490.diff
>> >
>> > —
>> > You are receiving this because your review was requested.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/genenetwork/genenetwork2/pull/490>, or unsubscribe
>> > <
>> https://github.com/notifications/unsubscribe-auth/AC55V7CJLNHDDX4QKX5MRXLSRVGSVANCNFSM4UCYV53Q
>> >
>> > .
>> >
>>
>>
>> --
>> Rob
>>
>> Robert W. Williams, Ph.D.
>> Chair: Department of Genetics, Genomics and Informatics
>> University of Tennessee Health Science Center
>> 71 S Manassas St, Memphis TN 38103-3310, TSRB Room 405
>> Tele/Cell: 901 604 4752
>> EMAIL: rwilliams@uthsc.edu or labwilliams@gmail.com
>>
>> —
>> You are receiving this because you were assigned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/genenetwork/genenetwork2/pull/490#issuecomment-733903592>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AANQJGDILPUBMSRVE324STLSRVKHVANCNFSM4UCYV53Q>
>> .
>>
>

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> I kind of wonder if it might make sense to have a "Rob testing site" or
> something where changes up for "Rob review" can be used.
>
> In the past this was less of an issue because I wasn't really using git
> correctly and was making all changes on the same local branch, but now that
> I'm actually separating them into separate branches it's difficult for Rob
> to test them if I'm currently using a different branch than the one with
> the changes.
>

If you pinged me, I could always set something up
on Penguin2. We have 3 open ports there, and right
now they aren't heavily utilized...

[...]

-- 
Bonface M. K. <https://www.bonfacemunyoki.com>
Chief Emacs Bazu / Rieng ya software sare
Mchochezi of: <https://upbookclub.com> / Twitter: @BonfaceKilz
GPG Key: D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

BonfaceKilz reviewed (approved):
This LGTM
